### ISSUE TITLE
NullPointerException in JPEGImageReader.read

### ISSUE THREAD

**Issue by THausherr** [haraldk/TwelveMonkeys#109]
I get this exception with the attached file:

    Exception in thread "main" java.lang.NullPointerException
	  at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.read(Unknown Source)
	  at javax.imageio.ImageIO.read(ImageIO.java:1448)
	  at javax.imageio.ImageIO.read(ImageIO.java:1308)
	  at JpegMassTest.main(JpegMassTest.java:20)

![dct-1425666923550](https://cloud.githubusercontent.com/assets/6665575/6537737/1b1e0dfa-c45a-11e4-9790-a4bebbaf9e25.jpg)
![dct-1425666923550 zip](https://cloud.githubusercontent.com/assets/6665575/6537748/3bbfeaf6-c45a-11e4-9ae1-d1dc4e988b57.png)

(testing attaching it both as a jpeg and as a zip)

**Comment by haraldk** [comment]
Hi,

I need to ask you again, are you sure you are using the latest 3.1-SNAPSHOT?

For this image I get a different exception and stack trace:

    javax.imageio.IIOException: Invalid JPEG file structure: SOS before SOF
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.readImageHeader(Native Method)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.readNativeHeader(JPEGImageReader.java:604)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.checkTablesOnly(JPEGImageReader.java:342)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.gotoImage(JPEGImageReader.java:476)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.readHeader(JPEGImageReader.java:597)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.getImageTypesOnThread(JPEGImageReader.java:778)
	  at com.sun.imageio.plugins.jpeg.JPEGImageReader.getImageTypes(JPEGImageReader.java:768)
	  at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.getImageTypes(Unknown Source)
	  at com.twelvemonkeys.imageio.plugins.jpeg.JPEGImageReader.main(Unknown Source)

**Comment by THausherr** [comment]
Weird thing: I only get e-mailed the comment, but not the edited comment. But yes, I am using 3.1-SNAPSHOT and I'm building it locally.

**Comment by THausherr** [comment]
I removed
     <compilerArgument>-g:lines</compilerArgument>
in the pom and seeing the java  line, it is likely what has happened (sof was null?).

**Comment by haraldk** [comment]
Hi again,

Another "issue" with the GitHub issue tracker, I guess.. It's rather simplistic. But it works okay-ish. Using Jira reminds me too much of my day job, so I'll stick to this for now. :-)

Anyway, this time it was my fault. I was testing with the image from 106 (which I still can't reproduce). Retesting with the attached image I found the problem. Yes, SOF == null.

Happy to inform that I just pushed a fix for this issue in 5a0c59604013520f2a1d79f9e04c231ee34cdb12. 


-- 
Harald K
